js/compiler.ts

```diff
@@ -186,6 +256,56 @@ export class DenoCompiler implements ts.LanguageServiceHost {
         : undefined;
   }
 
+  /**
+   * Returns a require that specifically handles the resolution of a transpiled
+   * emit of a dynamic ES `import()` from TypeScript.
+   */
+  private _makeLocalRequire(moduleMetaData: ModuleMetaData): AMDRequire {
+    const localRequire = (
+      deps: ModuleSpecifier[],
+      callback: AmdCallback,
+      errback: AmdErrback
+    ): void => {
+      log("localRequire", deps);
+      assert(
+        deps.length === 1,
+        "Local require requires exactly one dependency."
+      );
+      const [moduleSpecifier] = deps;
+      try {
+        const requiredMetaData = this.run(
+          moduleSpecifier,
+          moduleMetaData.fileName
+        );
+        callback(requiredMetaData.exports);
+      } catch (e) {
+        errback(e);
+      }
+    };
+    return localRequire;
+  }
+
+  /**
+   * Resolve the dependencies for a given module, but don't instantiate the
+   * module, but add the module to the instantiation stack.
+   */
+  private _resolveDependencies(moduleMetaData: ModuleMetaData): void {
```

Comment:
Consider calling this `runDependencies` or `instanciateDependencies` ?

Reply:
Actually this is neither run or instantiate... it really is "discover" or "collect" or "gather", in that none of the factories are run...  I am going to go with "gather".